lib/state_notifiers/token_notifier.dart

```diff
@@ -0,0 +1,481 @@
+      return;
+    }
+
+    if (oldIndex == newIndex) return;
+    //If the selected token moved down all other tokens between old and new index must decrease their sort index by 1 to move up
+    //If the selected token moved up all other tokens between old and new index must increase their sort index by 1 to move down
+    final selectedTokenMovedDown = newIndex > oldIndex;
+    if (selectedTokenMovedDown) newIndex--;
+    final reorderedTokens = <Token>[];
+    for (int i = min(oldIndex, newIndex); i <= max(oldIndex, newIndex); i++) {
+      final token = state[i];
+      if (i == oldIndex) {
+        Logger.info('Token ${token.label}: ${token.sortIndex} -> $newIndex');
+        reorderedTokens.add(token.copyWith(sortIndex: newIndex));
+        continue;
+      }
+
+      final newIndexThisToken = selectedTokenMovedDown ? i - 1 : i + 1;
+      Logger.info('Token ${token.label}: ${token.sortIndex} -> $newIndexThisToken');
+      reorderedTokens.add(token.copyWith(sortIndex: newIndexThisToken));
+    }
+
+    updateTokens(reorderedTokens);
+  }
+
+  void addTokenFromOtpAuth({required String otpAuth, required BuildContext context}) async {
+    Logger.info(
+      'Try to handle otpAuth:',
+      name: 'token_notifier.dart#addTokenFromOtpAuth',
+      error: otpAuth,
+    );
+
+    try {
+      Map<String, dynamic> qrParameterMap = parseQRCodeToMap(otpAuth);
+
+      Token? newToken = await _buildTokenFromMap(uriMap: qrParameterMap, uri: Uri.parse(otpAuth), context: context);
+
+      if (newToken == null) {
+        Logger.warning(
+          'Could not build token from qrParameterMap',
+          name: 'token_notifier.dart#addTokenFromQRCode',
+          error: qrParameterMap,
+        );
+        return;
+      }
+
+      if (newToken.pin != null && newToken.pin != false) {
+        Logger.info(
+          'New token has a pin. Locking token.',
+          name: 'token_notifier.dart#token_notifier',
+        );
+        newToken = newToken.copyWith(isLocked: true);
+      }
+
+      if (newToken is PushToken && state.contains(newToken)) {
+        showMessage(message: 'A token with the serial ${newToken.serial} already exists!', duration: Duration(seconds: 2), context: context);
+        return;
+      }
+
+      if (newToken is PushToken) {
+        _rollOutToken(newToken);
+      }
+
+      addToken(newToken);
+    } on ArgumentError catch (e) {
+      // Error while parsing qr code.
+      Logger.warning(
+        'Malformed QR code:',
+        name: 'main_screen.dart#_handleOtpAuth',
+        error: e.stackTrace,
+      );
+
+      showMessage(message: '${e.message}\n Please inform the creator of this qr code about the problem.', duration: Duration(seconds: 8), context: context);
+    }
+  }
+
+  void _enablePollingIfNeeded() {
+    if (globalRef?.read(settingsProvider).pollingEnabled == true) return;
+    if (state.whereType<PushToken>().isEmpty) return;
+
+    //enable polling on push token added
```

Comment:
space after //
I think this behavior (automatically enabling polling when a push token is added) should be removed, will make an issue for it